core/iwasm/libraries/lib-wasi-threads/test/stress_test_threads_creation.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2023 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <stdbool.h>
+
+enum CONSTANTS {
+    NUM_ITER = 90000,
+    NUM_RETRY = 5,
+    MAX_NUM_THREADS = 8,
+    RETRY_SLEEP_TIME_US = 4000,
+};
+
+unsigned int threads_executed = 0;
+unsigned int threads_creation_tried = 0;
+
+pthread_mutex_t mutex;
```

Comment:
```suggestion
pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
```

and then you can remove init/destroy calls in the main function.

Reply:
thanks